packages/arb-ts/src/lib/message/L2ToL1Message.ts

```diff
@@ -0,0 +1,374 @@
+import {
+  L2ToL1EventResult,
+  MessageBatchProofInfo,
+  OutgoingMessageState,
+} from '../dataEntities'
+import {
+  ARB_SYS_ADDRESS,
+  NODE_INTERFACE_ADDRESS,
+} from '../precompile_addresses'
+import networks, { Network } from '../networks'
+import { TransactionReceipt } from '@ethersproject/providers'
+import { Provider, Filter } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { BigNumber } from '@ethersproject/bignumber'
+import {
+  SignerOrProviderUtils,
+  SignerOrProvider,
+} from '../utils/signerOrProvider'
+import { wait } from '../utils/lib'
+
+import { Log } from '@ethersproject/abstract-provider'
+import { Outbox__factory } from '../abi/factories/Outbox__factory'
+import { IOutbox__factory } from '../abi/factories/IOutbox__factory'
+import { ArbSys__factory } from '../abi/factories/ArbSys__factory'
+import { hexZeroPad } from '@ethersproject/bytes'
+
+export class L2TransactionReceipt implements TransactionReceipt {
+  public readonly to: string
+  public readonly from: string
+  public readonly contractAddress: string
+  public readonly transactionIndex: number
+  public readonly root?: string
+  public readonly gasUsed: BigNumber
+  public readonly logsBloom: string
+  public readonly blockHash: string
+  public readonly transactionHash: string
+  public readonly logs: Array<Log>
+  public readonly blockNumber: number
+  public readonly confirmations: number
+  public readonly cumulativeGasUsed: BigNumber
+  public readonly effectiveGasPrice: BigNumber
+  public readonly byzantium: boolean
+  public readonly type: number
+  public readonly status?: number
+
+  constructor(tx: TransactionReceipt) {
+    this.to = tx.to
+    this.from = tx.from
+    this.contractAddress = tx.contractAddress
+    this.transactionIndex = tx.transactionIndex
+    this.root = tx.root
+    this.gasUsed = tx.gasUsed
+    this.logsBloom = tx.logsBloom
+    this.blockHash = tx.blockHash
+    this.transactionHash = tx.transactionHash
+    this.logs = tx.logs
+    this.blockNumber = tx.blockNumber
+    this.confirmations = tx.confirmations
+    this.cumulativeGasUsed = tx.cumulativeGasUsed
+    this.effectiveGasPrice = tx.effectiveGasPrice
+    this.byzantium = tx.byzantium
+    this.type = tx.type
+    this.status = tx.status
+  }
+
+  public getL2ToL1Events() {
+    const iface = ArbSys__factory.createInterface()
+    const l2ToL1Event = iface.getEvent('L2ToL1Transaction')
+    const eventTopic = iface.getEventTopic(l2ToL1Event)
+
+    const logs = this.logs.filter(log => log.topics[0] === eventTopic)
+
+    return logs.map(
+      log => (iface.parseLog(log).args as unknown) as L2ToL1EventResult
+    )
+  }
+
+  private getOutboxAddr(network: Network, batchNumber: BigNumber) {
+    // CHRIS: add the old network to the networks object, and look it up here
+    // CHRIS: null check? this shouldn't be possible
+    // CHRIS: disable linting by just using the batchnumber here
+    batchNumber
```

Comment:
This is where we'll want to use the batch number to check for the appopriate outbox, correct?

Reply:
Yep - I need to get the old outbox addresses from somewhere